backend/src/test/java/codezap/template/repository/ThumbnailRepositoryTest.java

```diff
@@ -0,0 +1,131 @@
+package codezap.template.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import codezap.category.domain.Category;
+import codezap.category.repository.CategoryRepository;
+import codezap.global.exception.CodeZapException;
+import codezap.global.repository.JpaRepositoryTest;
+import codezap.member.domain.Member;
+import codezap.member.repository.MemberRepository;
+import codezap.template.domain.SourceCode;
+import codezap.template.domain.Template;
+import codezap.template.domain.Thumbnail;
+
+@JpaRepositoryTest
+public class ThumbnailRepositoryTest {
```

Comment:
```suggestion
public class ThumbnailJpaRepositoryTest {
```
⚡ 
우리가 테스트를 하는 것은 `ThumbnailRepository` 가 아니라 `JpaRepository` 를 상속하는 `ThumbnailJpaRepository` 이죠.
테스트 하는 클래스를 명확하게 명시해 줄까요?

Reply:
저는 다음과 같이 생각하는데요. 

여기에 대해 짱수가 어떻게 생각하는지 궁금해요~

[관련 근거](https://github.com/woowacourse-teams/2024-code-zap/pull/626#discussion_r1753118059)

> 저는 짱수와 조금 다른 의견인데요.
> 
> 근거는 다음과 같습니다!
> 
> 1. 클래스 애너테이션에 JPA가 명시되어있음
> 2. 실제로 테스트 대상이 되는 객체는 JPA가 아닌 인터페이스를 사용하고 있음. SourceCodeJpaRepository가 아닌 SourceCodeRepository.
> 3. JPA를 사용하지 않게 되더라도 본 테스트의 본질은 유효함. DB와 데이터를 잘 주고받을 수 있는지! 만약 그렇게 된다면 애너테이션을 제거하고 메서드를 리팩토링하면 됨.

Reply:
링크 달아 주신 리뷰에도 답변 달아두었지만 보기 쉽게 여기도 옮겨 둘게요!

> 제가 근거를 더 자세히 적어둘 걸 그랬네요. 코멘트 여러번 왔다갔다 하게 해서 미안합니다 ㅠ
>
>제가 해당 부분에 코멘트를 작성한 이유는 다음과 같습니다.
>
>1. 테스트 작성 이유
>우리가 테스트를 작성한 이유 중에는 "JpaRepository 에서 제공하는, 메서드 명으로 만들어 주는 메서드들이 우리의 예상대로 동작하는지 확인하기 위함" 도 존재한다고 생각해요.
>즉, Jpa 라는 환경 때문에 작성되는 테스트라고 생각합니다.
>
>저의 경우 실제로 복잡한 JpaRepository 메서드 명을 사용하면 '제대로 작성한 것이 맞나?' 하는 불안함이 들 때가 있고, 이럴 때 테스트코드가 확신을 줄 수 있다고 생각해요.
>
>2. 빠르게 테스트 코드를 찾을 수 있다.
>아주 사소하지만, 테스트 이름을 클래스 이름과 동일하게 작성해 두면 클래스에서 cmd + shift + t 로 빠르게 테스트코드를 찾아갈 수 있습니다.
>추가되는 메서드에 대해서도 빠르게 테스트코드를 작성할 수 있게 될거에요.